src/main/java/com/lorepo/icplayer/client/module/choice/ChoicePrintable.java

```diff
@@ -21,49 +22,117 @@ public ChoicePrintable(ChoiceModel model, PrintableController controller) {
-		for (String optionHTML: optionHTMLs) {
+		for (String optionHTML: optionsHTMLs) {
 			result += optionHTML;
 		}
 		result += "</ol></div>";
-		
 		result = PrintableContentParser.addClassToPrintableModule(result, className);
-		
+
 		return result;
 	}
+
+	private List<String> getStyledOptionsIfPrintableState(boolean showAnswers, HashMap<String, String> printableState) {
+		ArrayList<ChoiceOption> orderedOptions = new ArrayList<ChoiceOption>();
+		ArrayList<Boolean> orderedIsSelectedOptions = new ArrayList<Boolean>();
+
+		List<String> optionHTMLs = new ArrayList<String>();
+
+		String optionState = printableState.get("options");
+		for (int i = 0; i < model.getOptionCount() &&  i < optionState.length(); i++) {
+			orderedOptions.add(model.getOption(i));
+			Boolean value = (optionState.charAt(i) == '1');
+			orderedIsSelectedOptions.add(value);
+		}
+		if (orderedIsSelectedOptions.size() != orderedOptions.size()) {
+			return optionHTMLs;
+		}
+
+		for (int i = 0; i < orderedOptions.size(); i++) {
+			ChoiceOption option = orderedOptions.get(i);
+			Boolean isDown = orderedIsSelectedOptions.get(i);
+			String optionHTML = createPrintableOptionIfSetState(option, showAnswers, isDown);
+			optionHTMLs.add(optionHTML);
+		}
+
+		return optionHTMLs;
+	}
+
+	private String createPrintableOptionIfSetState(ChoiceOption option, boolean showAnswers, Boolean isDown) {
+		String className = "";
+		String innerText = "";
+		String text = option.getText();
+		if (showAnswers) {
+			if (isDown) {
+				if (option.getValue() > 0) {
+					className = "ic_poption-down-correct-answer";
+				} else {
+					className = "ic_poption-down-wrong-answer";
+				}
+				innerText = "<div class=\"" + className + "\">" + text + "</div>";
+			} else {
+				innerText = text;
+			}
+		} else {
+			if (isDown) {
+				className = "ic_poption-down-answer";
+				innerText = "<div class=\"" + className + "\">" + text + "</div>";
+			} else {
+				innerText = text;
+			}
+		}
+		String optionHTML = "<li id=\"" + getOptionViewId(option) + "\">";
+		optionHTML += innerText;
+		optionHTML += "</li>";
+		return optionHTML;
+	}
+
+	private List<String> getStyledOptionsIfNotPrintableState(boolean showAnswers) {
+		ArrayList<ChoiceOption> orderedOptions = new ArrayList<ChoiceOption>();
+
+		List<String> optionHTMLs = new ArrayList<String>();
+
+		for (int i = 0; i < model.getOptionCount(); i++) {
+			orderedOptions.add(model.getOption(i));
+		}
+
+		for (ChoiceOption option: orderedOptions) {
+			String optionHTML = createPrintableOptionIfNotSetState(option, showAnswers);
+			optionHTMLs.add(optionHTML);
+		}
+		return optionHTMLs;
+	}
 	
-	private String createPrintableOption(ChoiceOption option, boolean showAnswers) {
-		String optionHTML = "";
+	private String createPrintableOptionIfNotSetState(ChoiceOption option, boolean showAnswers) {
+		String text = option.getText();
+		String className = "";
+		String innerText = "";
 		if (showAnswers && option.getValue() > 0) {

```

Comment:
brzydko - jak pozamykasz to w obiektach to sie uprosci

Reply:
Została zrobiona klasa odpowiedzialna za wykonywanie operacji w zależności od stanu.